### DISCUSSION TITLE
Use Multiple Provider in single screen

### DISCUSSION THREAD

**Discussion by Amanvein** [rrousselGit/riverpod#2908] [General]
To build the screen, I need to call multiple providers(generated) and multiple API's So different-different loader for each provider. What is the better way to manage it. (Riverpod 2.0).
@rrousselGit

**Comment by noriHanda** [comment]
@Amanvein I think it's pretty straight forward. You can follow [this document](https://docs-v2.riverpod.dev/docs/essentials/first_request#rendering-the-network-requests-response-in-the-ui:~:text=is%20first%20read.-,Rendering,-the%20network%20request%27s)

**Reply by Amanvein** [reply]
it is for single API call, what if UI build's with the multiple API and each API call have it's own provider

**Reply by noriHanda** [reply]
Just define as many providers as the API calls?
